torch/csrc/autograd/custom_function.h

```diff
@@ -120,6 +123,7 @@ struct TORCH_API AutogradContext {
   std::unordered_set<at::TensorImpl*> dirty_inputs_;
   std::vector<torch::autograd::SavedVariable> saved_variables_;
   variable_list to_save_;
+  bool materialize_grads_ = true;
```

Comment:
I'm not sure we want to put the default value here. It might be better in the constructor. The default initialization can have unexpected behavior in c++ so I usually prefer to be safe.